red/api/editor/sshkeys.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+var express = require("express");
+var os = require("os");
+var runtime;
+var settings;
+var needsPermission = require("../auth").needsPermission;
+
+function getUsername(userObj) {
+    var username = os.hostname();
```

Comment:
The hostname could change; for example when on DHCP. This is an issue users sometimes hit where we use the hostname to generate their flow file - a change of DHCP address leads to a different flow file being used.

We need to learn from that and not assume hostname is safe to use as a default value here.

I suggest, instead of `os.hostname()`, use a fixed string such as `_default_` - something unlike to be a user's real username.